### PR TITLE
Features/imperial units

### PR DESCRIPTION
**Pull request by merlos**
Implements the support of imperial units in ios and partially in apple watch. It implements #87 

In the iOS App it uses locale for determine the default. However he can force to use metric or imperial units through the Preferences screen.

In Apple Watch it uses locale settings to determine if has to use imperial or metric units. Ideally, both preferences should be synced, but we may need to create an AppGroup to share a default or send the new preference to the AppleWatch each time it is changed in the iOS App. I would consider this sync as a new PR/Feature.

I have to look further to this tutorial 
https://developer.apple.com/documentation/watchconnectivity/using_watch_connectivity_to_communicate_between_your_apple_watch_app_and_iphone_app

Also, I found out that we can share files just by adding the extension as part of the the target in the file Target membership option. This can be very useful for sharing logic between iOS App and the Extension.

Screenshot 2019-05-05 at 18 00 31

### CODE REVIEW

**Comment by vincentneo** [comment]
Didn't have much time to test out these days, but it appears that km doesn't work, and would result in a km value, say, 69.80km, expressed as 69.80mi, for the total tracked distance.

Will investigate if I've got the time.

Also, should anything below one mile be treated as feet for the total tracked distance? I don't know much about the imperial system (coming from a non imperial system country), but seeing 20m expressed as 0.01mi looks pretty weird.

**Comment by merlos** [comment]
Thanks @vincentneo 

I did a short test this afternoon: the phone was with metric and the watch with imperial. The speed apparently looked right... slightly smaller speed in the watch (but I did'nt do a proper scientific test).

Regarding the feet question, I am also more keen on metric system, but after your comment I asked an American colleague and he told me that miles looked OK.
 
Thanks, Juan

**Comment by vincentneo** [comment]
The speed appears to be fine, though it was something I have yet to test fully. 

To clarify my first comment, in case it was misunderstood as if the speed had the problem,
The issue happened when I tested it by loading a gpx file on map, and if the total tracked distance is below within meter range (e.g 250m) then its fine. If the total tracked distance is within kilometer range (e.g 69.80km), it would display as 69.80mi, despite imperial units being switched off, based on what I am observing on my iPhone.
If imperial is switched on, 69.80mi would of course be converted to 43.37mi.

TL;DR the km unit doesn't show for the total tracked distance.

Thank you.

**Comment by vincentneo** [comment]
I think I just fixed the bug that I described, it was a problem with `toKilometers()` method returning 'mi' instead of 'km', possibly from copy and paste?

Other than that simple-to-fix bug, I think its ready for merge.

**Comment by merlos** [comment]
Thanks @vincentneo for the fix.
